Adding Working Groups directory (#155)
moja-global/About_moja_global: working-groups/user-interface/About.md

> # User-interface (UI) working group

Reviewer: Add a line break

> Hello and welcome. This page talks about the UI working group. If you have any queries about the UI working group after going through this page please feel free to reach out to us on [slack.](https://join.slack.com/t/mojaglobal/shared_invite/zt-o6ta1ug0-rVLjAo460~d7JbZ~HpFFtw)

Reviewer: `This document outlines all the information about the User-interface (UI) working group. For any queries about the UI working group, please reach out to us [Slack](https://join.slack.com/t/mojaglobal/shared_invite/zt-o6ta1ug0-rVLjAo460~d7JbZ~HpFFtw)`.

> ## What’s a working group?

Reviewer: Give a line break

A working group is a group of people working together to achieve specified goals. The groups are domain-specific and focus on discussion or activity around a specific subject area.

> ## About the UI working group

Reviewer: Give a line break

<a href="#incubating"><img src="https://img.shields.io/static/v1?label=Status &message=Active &color=green" /></a>

The UI working group is responsible for developing the visual components, user interactions and user experience of the UI-based projects under moja global.

| Meeting details            |                                                                                                    |
| -------------------------- | -------------------------------------------------------------------------------------------------------------- |

> | Leads                      | [@Shloka Gupta](https://github.com/chicken-biryani?tab=followers) [@Soham Shah](https://github.com/sohamsshah) |

Reviewer: Why does your URL lead to individual followers list and not to your direct profile?

| Days                       | Every Saturday (19th Feb onwards)                                                                                                 |

> | Time                       | 10:30 (GMT) / 16:00  (IST) / 5:30 (EST)                                                  |

Reviewer: Add for AEST and CET as well

| On-going project           | Develop TSC working group notice board.                                                                        |
| Agenda                     | You can have a look at the upcoming agenda [here.](agenda.md)                                                  |

> | Meeting Notes              | Have a look at the Meeting notes over [here.](MeetingNotes.md)                                                 |

Reviewer: Avoid using `here`. Use proper hypertext here

| Contact                    | [Slack.](https://join.slack.com/t/mojaglobal/shared_invite/zt-o6ta1ug0-rVLjAo460~d7JbZ~HpFFtw)                 |

> | Relevant channels on Slack | #community-website, #general, #reporting-tool, #user-interface                                                 |

Reviewer: Remove `#general`

## What is the goal of this working group?

> We are trying to follow a “sprint-like” structure where on each “sprint” cycle we start with planning and then work towards the goals defined on that plan. You can know about the ongoing and upcoming work by attending our online meetings OR by looking at the agenda OR by going through the list of issues under the relevant repositories OR dropping a message on Slack.

Reviewer: If you are following a sprint, how are issues managed? GitHub Issues, Trello or Jira? Mention it here.

**The UI working group aims to accomplish the following in the coming days:**

> - **TSC working group notice board**. - There are several active and incubating working groups at moja global. We want to establish and develop a TSC working group notice, board. Please have a look at this [issue](https://github.com/moja-global/Google.Season.of.Documentation/issues/12) to know more.

Reviewer: Give a line break after every point.

- **FLINT-UI** - The FLINT-UI is a web-based user interface for FLINT for configuring simulations using the FLINT.Cloud APIs and run workflows on the browser itself. The FLINT-UI is a baby project and all contributions towards making it better are welcome. You can browse this [repository](https://github.com/moja-global/FLINT-UI) to learn more about FLINT-UI.
- **Reporting tool**. - The role of the reporting tool is to provide business intelligence for analyzing the FLINT output database and converting it into useful information and output. The current implementation of the reporting tool treats the Flux database as a simplified 2006 GL reporting table. We hope to refactor Angular components to be compatible with 13.x, add new features to the reporting tool, refactor old code and fix UI errors. If you want to contribute to the reporting tool, you can browse the issues in the [repository.](https://github.com/moja-global/FLINT.Reporting) If you need help chat with us on the [#reporting-tool](https://app.slack.com/client/T1G1M5HPF/C016YNZ2ALX) channel of Slack.

> ***Note:** The aim may change from time to time. Please have a look at the pinned message in the #user-interface channel in our Slack to know about the ongoing sprint.*

Reviewer: Bad markdown formatting here due to which all the text is in bold now.

## Tools and technologies

> The tools and techniques used vary from project to project. The most commonly employed are React, Angular, Vue, Sphinx, Docker, and Docusaurus. By browsing through the relevant repositories, you can learn more about which technologies are used where. Please note that it's not necessary for you to be familiar with these technologies to be able to contribute.

Reviewer: Add hyperlinks to all of the technologies.

**The UI working group aims to accomplish the following in the coming days:**

- **TSC working group notice board**. - There are several active and incubating working groups at moja global. We want to establish and develop a TSC working group notice, board. Please have a look at this [issue](https://github.com/moja-global/Google.Season.of.Documentation/issues/12) to know more.

- **Community website** - The Community website aims to serve a glimpse of moja global's projects and how they work to the wider community. The website hosts projects, case studies, blogs, tutorials, and more developed by the wider moja community to help people better understand our work. It’s currently under development and all contributions towards improving the project are welcome. With the help of the UI working group, we would like to make UI components for the community website, improve accessibility and usability, and design new pages and document UI development workflow. Have a look at the [repository](https://github.com/moja-global/community-website) to know more. If you feel stuck chat with us on the [#community-website](https://app.slack.com/client/T1G1M5HPF/C022V91MY0M) channel on Slack.

> - **FLINT-UI** - The FLINT-UI is a web-based user interface for FLINT for configuring simulations using the FLINT.Cloud APIs and run workflows on the browser itself. The FLINT-UI is a baby project and all contributions towards making it better are welcome. You can browse this [repository](https://github.com/moja-global/FLINT-UI) to learn more about FLINT-UI.

Reviewer 2: ```suggestion
- **FLINT-UI** - The FLINT-UI is a web-based user interface for FLINT, used to configure simulations with the FLINT.Cloud APIs, and run workflows in the browser itself. The FLINT-UI is a new project and all contributions towards making it better are welcome. For more information, see GitHub repository - [FLINT-UI](https://github.com/moja-global/FLINT-UI).
```

> - **Reporting tool**. - The role of the reporting tool is to provide business intelligence for analyzing the FLINT output database and converting it into useful information and output. The current implementation of the reporting tool treats the Flux database as a simplified 2006 GL reporting table. We hope to refactor Angular components to be compatible with 13.x, add new features to the reporting tool, refactor old code and fix UI errors. If you want to contribute to the reporting tool, you can browse the issues in the [repository.](https://github.com/moja-global/FLINT.Reporting) If you need help chat with us on the [#reporting-tool](https://app.slack.com/client/T1G1M5HPF/C016YNZ2ALX) channel of Slack.

Reviewer 2: "and converting it into useful information and output"
@chicken-biryani, could we say "into useful graphics and charts" instead? By saying "useful information" it can come across that the tool is reporting *not* useful information. :sweat_smile:

> **Note:** The aim may change from time to time. Please have a look at the pinned message in the #user-interface channel in our Slack to know about the ongoing sprint.

Reviewer 2: ```suggestion
**Note:** The sprint goal may change from time to time. Please have a look at the pinned message in the #user-interface channel in our Slack to find the goal for the ongoing sprint.
```

## Tools and technologies

> The tools and techniques used vary from project to project. The most commonly employed are [React](https://reactjs.org/), [Angular](https://angular.io/), [Vue](https://vuejs.org/), [Sphinx](https://www.sphinx-doc.org/en/master/),[Docker](https://www.docker.com/), and [Docusaurus](https://docusaurus.io/). By browsing through the relevant repositories, you can learn more about which technologies are used where. Please note that it's not necessary for you to be familiar with these technologies to be able to contribute.

Reviewer 2: ```suggestion
The tools and techniques used vary from project to project. The most commonly employed are:

 - [React](https://reactjs.org/)
 - [Angular](https://angular.io/)
 - [Vue](https://vuejs.org/)
 - [Sphinx](https://www.sphinx-doc.org/en/master/)
 - [Docker](https://www.docker.com/)
 - [Docusaurus](https://docusaurus.io/)
 
 By browsing through the relevant repositories, you can learn more about which technologies are used where. Please note, to be able to contribute it's not necessary for you to be familiar with these technologies from the start.
```

## How can I get involved?

- If you are interested in getting involved in this working group, there is a weekly meeting that takes place every Saturday at 10:30 (GMT) / 16:00  (IST) / 5:30 (EST). Feel free to stop by then (or any time!) to introduce yourself or send a message in the [#user-interface](https://app.slack.com/client/T1G1M5HPF/C010Z37GL2U) channel of our Slack.

> - If you’re interested in contributing, there is no need to wait for the meeting! As a part of the UI working group, there is an abundance of things one can work on. We value all kinds of contributions - not just code.

Reviewer 2: ```suggestion
- If you’re interested in contributing, there is no need to wait for the meeting! As a part of the UI working group, there is an abundance of things available to work on. We value all kinds of contributions, not just code.
```

> - We are particularly motivated to support new contributors and people who are looking to learn and develop their skills. Setting up the projects, trying them out, reporting bugs, working on good first issues, helping us build a stronger community are great ways to get started.

Reviewer 2: ```suggestion
- We are particularly motivated to support new contributors and people who are looking to learn and develop their skills. Setting up the projects, trying them out, reporting bugs, working on good first issues, and helping us build a stronger community are great ways to get started.
```

> - If working with GitHub seems like magic to you or you aren’t clear about the steps, look at our [GitHub Workflow](https://docs.moja.global/en/latest/DeveloperWorkflow/index.html) documentation. If you need any help, let us know. The community will be more than happy to help you out.

Reviewer 2: ```suggestion
- If working with GitHub seems like magic to you or you aren’t clear about the steps, look at our [GitHub Workflow](https://docs.moja.global/en/latest/DeveloperWorkflow/index.html) documentation. If you need any help, let us know. The community is more than happy to help you out.
```
